subsys/net/ip/utils.c

```diff
@@ -971,6 +971,168 @@ bool net_ipaddr_parse(const char *str, size_t str_len, struct sockaddr *addr)
 	return false;
 }
 
+const char *net_ipaddr_parse_mask(const char *str, size_t str_len,
+				  struct sockaddr *addr, uint8_t *mask_len)
+{
+	const char *next = NULL, *mask_ptr = NULL;
+	int parsed_mask_len = -1;
+	bool ret = false;
+
+	if (str == NULL || str_len == 0 || addr == NULL || mask_len == NULL) {
+		return NULL;
+	}
+
+	if (*str == '\0') {
+		return NULL;
+	}
+
+	for (int i = 0; i < str_len; i++) {
+		if (str[i] == ',') {
+			next = str + i + 1;
+			str_len = next - str - 1;
+			break;
+		}
+
+		if (str[i] == '/') {
+			mask_ptr = str + i;
+		}
+
+		if (str[i] == ' ') {
+			if (mask_ptr == NULL) {
+				mask_ptr = str + i;
+			}
+		}
```

Comment:
What does this condition do? Can the mask length be put after a space?

Reply:
Good catch. And the mask len is not suppose to be placed after a space, I will send a fix for this.

Reply:
There was no test for this specific scenario, and when I added one, the test checker did not catch it (there was an error in the checker).